frontend/src/metabase/lib/dom.js

```diff
@@ -320,6 +321,12 @@ export function open(
 ) {
   url = ignoreSiteUrl ? url : getWithSiteUrl(url);
 
+  // In the react sdk, we allow the host app to override how to open links
+  if (isEmbeddingSdk() && SDK_GLOBAL_PLUGINS.handleLink) {
+    SDK_GLOBAL_PLUGINS.handleLink(url);
+    return;
```

Comment:
I'm thinking about more ergonomic way of handling this. If I want to only handle one type of link, I don't want to have to deal with all the default values that Metabase does. This function kinda forces consumers to handle what they care and replicate the default values for other types of links they don't care about.

What if we force to return boolean as a way to indicate if the link is handled?

e.g.


```suggestion
    const isHandled = SDK_GLOBAL_PLUGINS.handleLink(url);
    if (isHandled) {
      return
    }
```

In their plugin
```ts
function handleLink(url) {
  if (isVideo(url)) {
    showVideoModal()
    return true
  }
}
```

Or we could have a similar to express [middleware pattern](https://expressjs.com/en/guide/using-middleware.html) using `next()` (we could literally use any other name).

```ts
function handleLink(url, next) {
  if (isVideo(url)) {
    return showVideoModal()
  }

  next()
}
```
I think it would be more ergonomic, the only catch is that they need to indicate whether they want Metabase to handle the rest of the links

I still think it's a bit better than forcing handling all other kind of links though. Let me know what you think?